File: public/img/2021-05-14-beauty/benchmark_numbers.ml

let rec generateNumbers i acc = if i = 0 then acc else string_of_int i :: acc

let nums = generateNumbers 1000000 []

let time f x =
  let t = Sys.time() in
  f x |> ignore;
  Printf.printf "Benchmark took %fµs\n" ((Sys.time() -. t) *. 1000000.);
  ()


let _ = time (List.fold_left (fun acc i -> i |> int_of_string |> ( + ) acc) 0) nums

File: public/img/2021-05-14-beauty/benchmark_numbers.go
package main

import (
	"fmt"
	"strconv"
	"time"
)

func main() {
	nums := generateNumbers(1000000)

	start := time.Now()

	sum := 0
	for _, v := range nums {
		intVal, _ := strconv.Atoi(v)
		sum += intVal
	}

	elapsed := time.Since(start)
	fmt.Printf("Benchmark took %s\n", elapsed)
}

func generateNumbers(number int) []string {
	nums := []string{}
	for i := 1; i <= number; i++ {
		nums = append(nums, strconv.Itoa(i))
	}

	return nums
}
